### ISSUE TITLE
Add fallback location to redshift.conf

### ISSUE THREAD

**Issue by jetojedno** [jonls/redshift#348]
If geoclue fails, redshift fails.  That's wrong.  There should be the ability to add a fallback (or default) location which is used if geoclue doesn't return a location.

See issue #333

**Comment by maandree** [comment]
I think these should be address in GeoClue. That way it is solved for all applications that use GeoClue.

I also suggest using [~/.config/geolocation](https://github.com/maandree/metar/blob/c11cdc62913fa167f804f99dc53e2264bed839b2/README#L77-L89) for this.

**Comment by maandree** [comment]
What? Couldn't you just tell GeoClue that you are fine with a fallback or last known location?

**Comment by jtarin47** [comment]
"redshift.config" is the most direct approach. Set it and forget it. Only drawback is if your a frequent long-distance traveler.

**Comment by jetojedno** [comment]
@maandree funnily, I don't think I'm in a position to tell the developers / maintainers / architects of a package what to do with their package.  I can request / suggest changes, but they take responsibility for the system and and changes.  The same with redshift.
@jtarin47 I do travel a lot.  Most of the time I have internet once or twice a day ...

Having a separate request (or separate part of the response) where GeoClue provides that last known good location would work on both sides.  Using the new info would require changes to redshift, but any acceptable solution is going to.

**Comment by jetojedno** [comment]
@jtarin47 that solution doesn't work if you travel a lot, as I do.  If you're (relatively) static then you don't need GeoClue at all.

@maandree  I've requested a satisfactory compromise https://bugs.freedesktop.org/show_bug.cgi?id=96861

**Comment by ksinkar** [comment]
I think a better way would be use the TimeZone locale to be used of GeoClue is not available or disabled. Each timezone is always passes through a city and if you are in that timezone, it approximates your day-night cycle. And every Linux machine always has a timezone associated with it. If you travel a lot, then you could simply put your timezone on autosync mode and it would still solve your problem.

**Comment by ksinkar** [comment]
I think a third location provider can be added called `system-timezone` which can solve this problem. So people can choose to that. Also this should be the default, because system-timezone shall always be present whether geoclue is present or not. This is something that most kernels give by default.

**Comment by rugk** [comment]
But it makes a difference if you are in the same timezone near the equator or near the north or south pole.

**Comment by ksinkar** [comment]
@rugk, I agree with you. `system-timezone` is definitely not a silver bullet that solves all the problems. If someone is not okay with using `system-timezone`, they can always go for Geoclue or manual  setting, but for the program to start and then crash because Geoclue has not been enabled can be a big turnoff for first-time users. The program  should just run when it starts with sensible defaults and without any configuration required or for anyone to come here and read the README. Once the program has started a user should be able to customise the program to his or her own preferences. But if the program  fails on startup itself, users will think it is a buggy FLOSS and simply not use it.

`system-timezone` is a sensible default; nothing more and nothing less.

**Comment by mirh** [comment]
Isn't this the same of #36?
Also, geoclue issue moved here now
https://gitlab.freedesktop.org/geoclue/geoclue/issues/51
